tests/files/c_source/enums.h

```diff
@@ -0,0 +1,191 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018-2019 Erik Moqvist
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/**
+ * This file was generated by cantools version 32.6.1 Thu Jan 24 16:45:28 2019.
+ */
+
+#ifndef ENUMS_H
+#define ENUMS_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#ifndef EINVAL
+#    define EINVAL 22
+#endif
+
+#define ENUMS_VEHICLE_STATUS_FRAME_ID (0x00u)
+
+typedef enum {
+    OFF = 0,
+    NOMINAL = 1,
+    OVERHEAT = 2
+} enums_inverter_state;
+
+typedef enum {
+    OK = 0,
+    FAULTED = 1
+} enums_wheel_speed_sensor_state;
```

Comment:
There is a high risk of name clashes without the `<database>_<message>_<signal>_`-prefix on all enums and their values. With prefixes the generated enums would look like:

``` c
enum enums_vehicle_status_inverter_state_choice_t {
    enums_vehicle_status_inverter_state_off_choice_t = 0,
    enums_vehicle_status_inverter_state_nominal_choice_t = 1,
    enums_vehicle_status_inverter_state_overheat_choice_t = 2
};

enum enums_vehicle_status_wheel_speed_sensor_state_choice_t {
    enums_vehicle_status_wheel_speed_sensor_state_ok_choice_t = 0,
    enums_vehicle_status_wheel_speed_sensor_state_faulted_choice_t = 1
};
```

In some systems only a subset of the signal values are named choices and the rest are an integer. For example, a 8 bits signal may name value 0 "NoChange" and value 1 "Error", while values 2-255 represents some voltage. I've absolutely no idea how common this is, but I've certainly seen it. Generating defines instead of enums is more flexible.

What is the reason for generating enums instead of defines?

Reply:
- excellent point on the clashes, that's easy enough to add
- using just a subset of signals as named choices is a little trickier to address, I'm not too sure what to do about this.
- the rational behind this change is that it tightens up interfaces. When using this generated code, it will be more clear what values a given member variable in a struct should accept, which leads to more readable, understandable and debuggable code, and reduces errors that just using a raw value introduces. This change would likely propagate outwards as well, as it would allow the passing of enums (which have their meaning and restrictions inherent in the type) instead of just raw values when working with functions manipulating CAN messages.

Reply:
The enums must have an additional value for invalid unpacked values, or something similar.

Reply:
Hmm. At this point it's looking like this is more trouble then it's worth. I'm going to close this PR, sorry for wasting your time!

Reply:
Ok!